File: 3 - atividade TAD/javaa/gerenciamentoDeBiblioteca/Livro.java
package javaa.gerenciamentoDeBiblioteca;

public class Livro{
    private String titulo;
    private String autor;
    private String anoPublicacao;
    private Integer quantidadeCopia;

    public Livro(String titulo,String autor,String anoPublicacao,Integer quantidadeCopia){
        this.titulo = titulo;
        this.autor = autor;
        this.anoPublicacao = anoPublicacao;
        this.quantidadeCopia = quantidadeCopia;
    }

    public void obterInformacao(){
        System.out.println("O título do livro é "+ titulo);
        System.out.println("O autor do livro é "+ autor);
        System.out.println("O ano de publicação do livro é "+ anoPublicacao);
    }

    public void verificarTemCopia(){
        if(quantidadeCopia >= 0)
            System.out.println("Ainda tem "+ quantidadeCopia+ " cópias disponíveis");
        else
            System.out.println("Não tem mais copias disponíveis");
    }
    
    public void emprestaCopia(){quantidadeCopia--;};
    public void devolveCopia(){quantidadeCopia++;};
}

File: 3 - atividade TAD/haskell/gerenciamentoBiblioteca/livro.hs
newtype Livro = Livro (String, String, String, Int)

inicializaLivro:: String->String->String->Int->Livro
inicializaLivro titulo autor anoPublicacao quantidadeCopia = Livro(titulo, autor, anoPublicacao, quantidadeCopia)

obterInformacao :: Livro -> IO()
obterInformacao(Livro (titulo,autor,anoPublicacao,_)) = do
    putStrLn $ "O titulo do livro é " ++show titulo
    putStrLn $ "O autor do livro é " ++show autor
    putStrLn $ "O ano de publicação do livro é "++ show anoPublicacao

verificarTemCopia :: Livro ->IO()
verificarTemCopia(Livro (_,_,_,quantidadeCopia)) = do
    if quantidadeCopia > 0 
    then putStrLn $ "Ainda tem "++ show quantidadeCopia ++ " cópias disponíveis"
    else putStrLn $ "Não tem mais cópias disponíveis"

emprestaCopia :: Livro -> Livro
emprestaCopia (Livro (titulo,autor,anoPublicacao,quantidadeCopia)) = Livro(titulo,autor,anoPublicacao,quantidadeCopia-1)

devolveCopia :: Livro -> Livro
devolveCopia (Livro (titulo,autor,anoPublicacao,quantidadeCopia)) = Livro(titulo,autor,anoPublicacao,quantidadeCopia+1)
